lenskit-core/src/main/java/org/grouplens/lenskit/data/event/LikeBatch.java

```diff
@@ -20,18 +20,102 @@
  */
 package org.grouplens.lenskit.data.event;
 
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
 /**
  * Generic interface representing a batch of unary interactions.  Events of this type will generally
  * not have timestamps.  An example of a use of this event type is to represent play counts in a
  * summarized data set.
+ * <p>
+ * Like batches can be constructed with {@link LikeBatchBuilder} or {@link #create(long, long, int)}.
  *
  * @since 2.2
  * @see Like
  * @see Events#likeBatch(long, long, int)
  */
-public interface LikeBatch extends Event {
+public class LikeBatch implements Event {
+    private final long userId;
+    private final long itemId;
+    private final int count;
+
+    LikeBatch(long uid, long iid, int ct) {
+        userId = uid;
+        itemId = iid;
+        count = ct;
+    }
+
+    public static LikeBatchBuilder newBuilder() {
+        return new LikeBatchBuilder();
+    }
+
+    public static LikeBatch create(long user, long item, int count) {
+        return new LikeBatch(user, item, count);
+    }
+
+    @Override
+    public long getUserId() {
+        return userId;
+    }
+
+    @Override
+    public long getItemId() {
+        return itemId;
+    }
+
     /**
-     * Get the number of likes
+     * Get the number of items liked.
+     * @return The number of items liked.
      */
-    int getCount();
+    public int getCount() {
```

Comment:
Mostly because public fields are, in general, a code smell, and I don't see a whole lot of cost to just never using them rather than allowing them in certain special cases. There's no run-time overhead to speak of; the JIT will very happily inline `getCount()`, especially since we are making these classes final and concrete (the best-case scenario for method inlining).